video/video.go

```diff
@@ -84,8 +84,21 @@ func (v *Video) RenderDraw(drw system.Drawer, op draw.Op) {
 		unchanged = false
 	}
 	v.framePlayed++
-	bb := v.Geom.TotalBBox.Add(v.Scene.SceneGeom.Pos)
-	drw.Scale(bb, frame, image.Rectangle{}, v.Rotation, draw.Src, unchanged)
+	tbb := v.Geom.TotalBBox
+	bb := tbb.Add(v.Scene.SceneGeom.Pos)
+	ibb := frame.Bounds()
+	pos := v.Geom.Pos.Total.ToPoint()
+	if pos.X < tbb.Min.X { // scrolled off left
+		ibb.Min.X = tbb.Min.X - pos.X
+	}
+	if pos.Y < tbb.Min.Y {
+		ibb.Min.X = tbb.Min.Y - pos.X
+	}
+	sz := ibb.Size()
+	if sz.X <= 0 || sz.Y <= 0 {
+		return
+	}
+	drw.Scale(bb, frame, ibb, v.Rotation, draw.Src, unchanged)
```

Comment:
Is there a way to avoid forcing each widget to do all of this work manually?

Reply:
making a helper function..  could do that.